hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/SnapshotDeletingService.java

```diff
@@ -182,13 +199,18 @@ public BackgroundTaskResult call() throws Exception {
             String snapshotBucketKey = dbBucketKey + OzoneConsts.OM_KEY_PREFIX;
             iterator.seek(snapshotBucketKey);
 
-            while (deletedIterator.hasNext()) {
+            int deletionCount = 0;
+            while (deletedIterator.hasNext() &&
+                deletionCount <= keyLimitPerSnapshot) {
               Table.KeyValue<String, RepeatedOmKeyInfo>
                   deletedKeyValue = deletedIterator.next();
               String deletedKey = deletedKeyValue.getKey();
 
               // Exit if it is out of the bucket scope.
               if (!deletedKey.startsWith(snapshotBucketKey)) {
+                // If snapshot deletedKeyTable doesn't have any
+                // entry in the snapshot scope it can be reclaimed
+                purgeSnapshotKeys.add(snapInfo.getTableKey());
                 break;
               }
```

Comment:
We haven't checked `deletedDirTable` emptiness right? Or add a TODO here

Reply:
We should add once we are done with changes in the `DirectoryDeletingService`, will update the PR.